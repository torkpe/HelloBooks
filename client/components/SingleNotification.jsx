import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';

import { getNotification } from '../actions/notification';

/**
 * @description Renders each notification
 * 
 * @param {object} props
 * 
 * @return {XML} JSX
 */
const SingleNotification = (props) => {
  const { notification, userId } = props;
  const mdlLink = `
  mdl-button
  mdl-button--colored
  mdl-js-button
  mdl-js-ripple-effect notifyButton
  `;
  return (
    <tbody>
      <tr>
        <td className="notifications">
          {notification.message}
          {notification.bookId ?
            <span className="notifyLink">
              <Link
              to={`/single/${notification.bookId}`}
              >
                  View Book
              </Link>
            </span>
          : ''}
        </td>
      </tr>
    </tbody>
  );
};
export default SingleNotification;
