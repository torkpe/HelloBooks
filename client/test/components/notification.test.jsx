import React from 'react';
import toJson from 'enzyme-to-json';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import SingleNotification from '../../components/Notification.jsx';
import storeConfiguration from '../../store/store';

configure({ adapter: new Adapter() });

describe('<SingleNotification />', () => {
  const props = {
    notification: {
      message: 'eze goes to school',
      bookId: 1
    },
  };
  it('renders <SingleNotification /> component', () => {
    const shallowComponent = shallow(<SingleNotification {...props} />);
    const tree = toJson(shallowComponent);
    expect(tree).toMatchSnapshot();
  });
});
