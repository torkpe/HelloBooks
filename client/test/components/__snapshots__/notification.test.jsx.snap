// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleNotification /> renders <SingleNotification /> component 1`] = `
<tbody>
  <tr>
    <td
      className="notifications"
    >
      eze goes to school
      <span
        className="notifyLink"
      >
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/single/1"
        >
          View Book
        </Link>
      </span>
    </td>
  </tr>
</tbody>
`;
